delaymodel.py

```diff
@@ -229,18 +236,46 @@ def calculate_delay(self, publish : bool = True):
         """
         while True:
             redis_publish_service_pulse(self.redis_obj, SERVICE_NAME)
-            loadtimes_on_hand = list(self.self.source_pointings_dict.keys())
-            if self.source_pointing_update.is_set():
-                #A new pointing has just arrived. Its loadtime must be checked and if within the 
-                #threshold, delay coefficients must be calculated for it and sent to the F-Engines.
-                new_loadtimes = list(set(loadtimes_on_hand) ^ set(list(self.self.source_pointings_dict.keys())))
-                new_loadtimes_sub_threshold = [loadtime for loadtime in new_loadtimes if loadtime <= self.threshold and loadtime > 0]
-
-
-            else:    
-                t = np.floor(time.time())
-                tts = [3, (TIME_INTERPOLATION_LENGTH/2) + 3, TIME_INTERPOLATION_LENGTH + 3] # Interpolate time samples with 3s advance
-                tts = np.array(tts) + t
+
+            #Now, we want to check whether the source pointing has updated OR whether it has been over 5 seconds since
+            #we last calculated delay coefficients:
+            t = time.time()
+            if ((t - self.delay_data["loadtime"]) >= 5 or self.source_point_update.is_set()):
+                #clear event here to avoid risk that the listening process sets the event as loadtime becomes > 5 and we miss a message.
+                self.source_point_update.clear()
```

Comment:
I think this comment is misleading -- you wouldn't miss a message if the flag was set while this thread is processing -- you'd simply process the command on the next loop.

I don't know how python deals with threading on dictionary accesses, but I'm not convinced any of this code actually handles `source_pointings_dict` properly -- items are pulled out of this dict one at a time, and I don't think anything prevents the listening thread from changing the values. I think any reading or writing of the dict would require a lock on it

Reply:
Bit wordy but I'll change it to something like:
```
#clear event here as opposed to at the end of the if block
#to reduce risk that the listening process sets the event as loadtime becomes > 5 and we 
#only process the new pointing ~5s from now or when the next pointing arrives.
```

Reply:
Thanks, this now actually makes sense to me.